extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisorIOConfig.java

```diff
@@ -0,0 +1,178 @@
+{
+  private final String endpoint;
+
+  // In determining a suitable value for recordsPerFetch:
+  //   - Each data record can be up to 1 MB in size
+  //   - Each shard can read up to 2 MB per second
+  //   - The maximum size of data that GetRecords can return is 10 MB. If a call returns this amount of data,
+  //     subsequent calls made within the next 5 seconds throw ProvisionedThroughputExceededException.
+  //
+  // If there is insufficient provisioned throughput on the shard, subsequent calls made within the next 1 second
+  // throw ProvisionedThroughputExceededException. Note that GetRecords won't return any data when it throws an
+  // exception. For this reason, we recommend that you wait one second between calls to GetRecords; however, it's
+  // possible that the application will get exceptions for longer than 1 second.
+  private final Integer recordsPerFetch;
+  private final Integer fetchDelayMillis;
+
+  private final String awsAccessKeyId;
+  private final String awsSecretAccessKey;
+  private final String awsAssumedRoleArn;
+  private final String awsExternalId;
+  private final boolean deaggregate;
+
+  @JsonCreator
+  public KinesisSupervisorIOConfig(
+      @JsonProperty("stream") String stream,
+      @JsonProperty("endpoint") String endpoint,
+      @JsonProperty("region") KinesisRegion region,
+      @JsonProperty("replicas") Integer replicas,
+      @JsonProperty("taskCount") Integer taskCount,
+      @JsonProperty("taskDuration") Period taskDuration,
+      @JsonProperty("startDelay") Period startDelay,
+      @JsonProperty("period") Period period,
+      @JsonProperty("useEarliestSequenceNumber") Boolean useEarliestSequenceNumber,
+      @JsonProperty("completionTimeout") Period completionTimeout,
+      @JsonProperty("lateMessageRejectionPeriod") Period lateMessageRejectionPeriod,
+      @JsonProperty("earlyMessageRejectionPeriod") Period earlyMessageRejectionPeriod,
+      @JsonProperty("recordsPerFetch") Integer recordsPerFetch,
+      @JsonProperty("fetchDelayMillis") Integer fetchDelayMillis,
+      @JsonProperty("awsAccessKeyId") String awsAccessKeyId,
+      @JsonProperty("awsSecretAccessKey") String awsSecretAccessKey,
+      @JsonProperty("awsAssumedRoleArn") String awsAssumedRoleArn,
+      @JsonProperty("awsExternalId") String awsExternalId,
+      @JsonProperty("deaggregate") boolean deaggregate
+  )
+  {
+    super(
+        Preconditions.checkNotNull(stream, "stream"),
+        replicas,
+        taskCount,
+        taskDuration,
+        startDelay,
+        period,
+        useEarliestSequenceNumber,
+        completionTimeout,
+        lateMessageRejectionPeriod,
+        earlyMessageRejectionPeriod
+    );
+    this.endpoint = endpoint != null
+                    ? endpoint
+                    : (region != null ? region.getEndpoint() : KinesisRegion.US_EAST_1.getEndpoint());
+    this.recordsPerFetch = recordsPerFetch != null ? recordsPerFetch : 4000;
```

Comment:
Use KinesisIOConfig.DEFAULT_RECORDS_PER_FETCH and KinesisIOConfig.DEFAULT_FETCH_DELAY_MILLIS